File: crud-item-list/src/main/java/com/gurunars/crud_item_list/CheckerSolidChunk.java
package com.gurunars.crud_item_list;

import java.util.List;

import java8.util.function.Function;

final class CheckerSolidChunk implements Function<List<Integer>, Boolean> {

    @Override
    public Boolean apply(List<Integer> positions) {
        if (positions.size() == 0) {
            return false;
        }

        for (int i = 1; i < positions.size(); i++) {
            if(positions.get(i) - positions.get(i-1) != 1) {
                return false;
            }
        }
        return true;
    }

}

File: crud-item-list/src/main/kotlin/com/gurunars/crud_item_list/CheckerSolidChunk.kt
package com.gurunars.crud_item_list

import java8.util.function.Function

internal class CheckerSolidChunk : Function<List<Int>, Boolean> {

    override fun apply(positions: List<Int>) =
        positions.isNotEmpty() &&
        (1..positions.size - 1).none { positions[it] - positions[it - 1] != 1 }

}
